cocotb/run.py

```diff
@@ -0,0 +1,929 @@
+# Copyright cocotb contributors
+# Licensed under the Revised BSD License, see LICENSE for details.
+# SPDX-License-Identifier: BSD-3-Clause
+
+import subprocess
+import os
+import sys
+import tempfile
+import re
+import cocotb
+import logging
+import shutil
+from xml.etree import cElementTree as ET
+import threading
+import signal
+import cocotb._vendor.find_libpython as find_libpython
+import cocotb.config
+
+from distutils.spawn import find_executable
+from distutils.sysconfig import get_config_var
```

Comment:
`distutils` is begin [deprecated](https://www.python.org/dev/peps/pep-0632/). `distutils.sysconfig.get_config_var` can be replace with `sysconfig.get_config_var`. `find_executable` can be replaced with `shutil.which`.